File: offer/go/03_1.go
package main

import "fmt"

func findRepeatNumber(nums []int) int {
	m := make(map[int]int)
	len := len(nums)
	for i := 0; i < len; i++ {
		if m[nums[i]] != 0 {
			return nums[i]
		}
		m[nums[i]]++
	}
	return 0
}

func main() {
	nums := []int{2, 3, 1, 0, 2, 5, 3}
	res := findRepeatNumber(nums)
	fmt.Println(res)
}

File: offer/cpp/03_1.cpp
#include <iostream>
#include <map>
#include <vector>
using namespace std;

class Solution {
public:
  int findRepeatNumber(vector<int> &nums) {
    map<int, int> map;
    for (const auto &num : nums) {
      if (map.count(num)) {
        return num;
      }
      map[num] = 1;
    }
    return 0;
  }
};

int main(int argc, const char *argv[]) {
  vector<int> nums{2, 3, 1, 0, 2, 5, 3};
  auto res = (new Solution())->findRepeatNumber(nums);
  cout << res << endl;
  return 0;
}
